cinema/views.py

```diff
@@ -131,8 +147,14 @@ def get_queryset(self):
     def get_serializer_class(self):
         if self.action == "list":
             return OrderListSerializer
-
         return OrderSerializer
 
     def perform_create(self, serializer):
         serializer.save(user=self.request.user)
+
+    def get_permissions(self):
+        if self.action == "create":
+            self.permission_classes = (IsAuthenticated,)
+        else:
+            self.permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
+        return [permission() for permission in self.permission_classes]
```

Comment:
why do u need this? u can use IsAuthenticated for all methods here, cause list is in SAFE_METHODS

Reply:
Actually I was thinking about other methods, pretty long time, till I mentioned parent classes)
Should be more scrupulous for sure.